### Question 1: Given some sample data, write a program to answer the following: click here to access the required data set
### On Shopify, we have exactly 100 sneaker shops, and each of these shops sells only one model of shoe. We want to do some analysis of the average order value (AOV). When we look at orders data over a 30 day window, we naively calculate an AOV of USD 3145.13. Given that we know these shops are selling sneakers, a relatively affordable item, something seems wrong with our analysis.

### a) Think about what could be going wrong with our calculation. Think about a better way to evaluate this data. 


In [1]:
import pandas as pd
import numpy as np

In [6]:
data = pd.read_csv("2019 Winter Data Science Intern Challenge Data Set.csv")

In [14]:
data.head()

order_id  shop_id  user_id  order_amount  total_items payment_method  \
0         1       53      746           224            2           cash   
1         2       92      925            90            1           cash   
2         3       44      861           144            1           cash   
3         4       18      935           156            1    credit_card   
4         5       18      883           156            1    credit_card   

        created_at  
0  3/13/2017 12:36  
1   3/3/2017 17:38  
2   3/14/2017 4:23  
3  3/26/2017 12:43  
4    3/1/2017 4:35

In [15]:
data.shape

(5000, 7)

When we think of 'Average' in Average Order Value (AOV), most of times we relate it to only one measure of central tendency ie is mean. But we also need to check for the other two - mean and mode. So let us check all the three in regards to this dataset

In [24]:
df = pd.DataFrame(data)

df['order_amount'].mean()


3145.128

In [25]:
df['order_amount'].median()

284.0

In [26]:
df['order_amount'].mode()

0    153
dtype: int64

In [27]:
df['order_amount'].describe()

count      5000.000000
mean       3145.128000
std       41282.539349
min          90.000000
25%         163.000000
50%         284.000000
75%         390.000000
max      704000.000000
Name: order_amount, dtype: float64

we can see from the above analysis that standard deviation is way to much from the mean. So we need to choose AOV using either median or mode.

### b. What metric would you report for this dataset?

As we saw the deviation is way to much in this dataset and the possible cause would be the outliers. In order to eliminate the influence of the outliers we need to go with <strong> the mode as a metric</strong>. Also, modal value represent the most frequent price of the shoes which says alot about the buyers buy most shoes at the modal price than other.

### c. What is its value?

From the analysis in the part a) - <strong> Mode = 153 </strong>

### Question 2: For this question you’ll need to use SQL. Follow this link to access the data set required for the challenge. Please use queries to answer the following questions. Paste your queries along with your final numerical answers below.

### a) How many orders were shipped by Speedy Express in total?

SELECT COUNT(DISTINCT Orders.OrderID) as Speedy_orders  
FROM Orders  
JOIN Shippers on Orders.ShipperID = Shippers.ShipperID  
WHERE ShipperName = 'Speedy Express';  

Output - <b> Speedy_orders 54 </b>

### b) What is the last name of the employee with the most orders?

SELECT LastName, COUNT(DISTINCT OrderID) as No_of_Orders  
FROM Orders  
JOIN Employees USING(EmployeeID)  
GROUP BY EmployeeID  
ORDER BY No_of_Orders DESC  
LIMIT 1  

Output - <b> Peacock 40 </b>

### c) What product was ordered the most by customers in Germany?

SELECT ProductName, SUM(Quantity) as ProductQuantity  
FROM OrderDetails  
JOIN Orders USING(OrderID)  
JOIN Customers USING(CustomerID)  
JOIN Products USING(ProductID)  
WHERE Country = 'Germany'
GROUP BY ProductID  
ORDER BY ProductQuantity DESC  
LIMIT 1  

Output - <b> Boston Crab Meat 160 </b>